### PR TITLE
chore(deps): Bumps JavaScript SDK to v8.41.0

### PR DESCRIPTION
**Pull request by antonis**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

Bumps JavaScript SDK to v8.41.0 and [fixes broken tests](https://github.com/getsentry/sentry-react-native/pull/4351/commits/3f1e81713ff34345d37828c0d19031b554d3816a) ([see this comment for context](https://github.com/getsentry/sentry-react-native/pull/4325#issuecomment-2505728868))
See https://github.com/getsentry/sentry-react-native/pull/4325 for the changelog.

## :bulb: Motivation and Context

Bumping https://github.com/getsentry/sentry-react-native/pull/4325 from 8.41.0 to 8.42.0 introduced new breaking changes. This PR attempts to [handle the 8.41.0 changes 1st for simplicity](https://github.com/getsentry/sentry-react-native/pull/4325#issuecomment-2505728868).

## :green_heart: How did you test it?
CI, Manually

## :pencil: Checklist

- [ ] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] I updated the docs if needed.
- [x] I updated the wizard if needed.
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

https://github.com/getsentry/sentry-react-native/pull/4325

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -49,6 +49,9 @@
 - Bump Android SDK from v7.18.0 to v7.18.1 ([#4329](https://github.com/getsentry/sentry-react-native/pull/4329))
   - [changelog](https://github.com/getsentry/sentry-java/blob/main/CHANGELOG.md#7181)
   - [diff](https://github.com/getsentry/sentry-java/compare/7.18.0...7.18.1)
+- Bump JavaScript SDK from v8.40.0 to v8.41.0 ([#4351](https://github.com/getsentry/sentry-react-native/pull/4351))
```
Comment (krystofwoldrich) [inline CHANGELOG.md:52]: Let's add small `### Changes` section and add `- Falsy values of `options.environment` (empty string, undefined...) default to `production``

**Review by krystofwoldrich — APPROVED** [review]
LG 🚀 Let's just add a small note to change log.

File: CHANGELOG.md (hunk 1 above)
Comment (antonis) [reply CHANGELOG.md:57]: Good point 👍 
thank you for catching this and adding a changelog @krystofwoldrich 🙇

**Comment by antonis** [comment]
> Thank you for making the JS bumps incrementally, this makes it super nice to follow the changes and to review it.

I'll follow with [another PR for the 8.42.0 breaking changes](https://github.com/getsentry/sentry-react-native/pull/4325#issuecomment-2528487799) before we proceed with 8.43.0. The JS SDK is moving fast 😅  🚀

**Pull request merged**